pandas/core/dtypes/cast.py

```diff
@@ -246,6 +246,99 @@ def trans(x):
     return result
 
 
+def maybe_cast_result(
+    result, obj: ABCSeries, numeric_only: bool = False, how: str = ""
+):
+    """
+    Try casting result to a different type if appropriate
+
+    Parameters
+    ----------
+    result : array-like
+        Result to cast.
+    obj : ABCSeries
+        Input series from which result was calculated.
+    numeric_only : bool, default False
+        Whether to cast only numerics or datetimes as well.
+    how : str, default ""
+        How the result was computed.
+
+    Returns
+    -------
+    result : array-like
+        result maybe casted to the dtype.
+    """
+    if obj.ndim > 1:
+        dtype = obj._values.dtype
+    else:
+        dtype = obj.dtype
+    dtype = maybe_cast_result_dtype(dtype, how)
+
+    if not is_scalar(result):
+        if is_extension_array_dtype(dtype) and dtype.kind != "M":
+            # The function can return something of any type, so check
+            #  if the type is compatible with the calling EA.
+            # datetime64tz is handled correctly in agg_series,
```

Comment:
this comment is now out-of-context, can you add a TODO to fix this

Reply:
I will change this to:

````
# The result may be of any type, cast back to original
# type if it's compatible.
````

The latter half of this comment was explaining why the special handling of datetime64tz was removed in [this PR](https://github.com/pandas-dev/pandas/commit/e1cadfa289c765978b812fc39e6af3e8d209d172), but I think is okay to remove.